### ISSUE TITLE
[json] Configure the severity level of trailing commas

### ISSUE THREAD

**Issue by borekb** [Microsoft/vscode#19992]
VS Code already supports comments in config JSON as per https://github.com/Microsoft/vscode/issues/1341 which is very useful. It also means that the `.json` files are no longer JSON files, strictly speaking, so while you're at it, it would also be great if I didn't need to juggle the trailing commas all the time. For example, these could be taken as valid "JSONs":

```json
[1, 2,]
```

```json
{
    "key1": 1,
    "key2": true,
}
```

I understand the controversy behind this, just posting this after having spent too much time fixing commas in various config files :)

**Comment by bpasero** [comment]
@borekb careful though, this is for our settings story and is unrelated to general JSON files.

**Comment by borekb** [comment]
@bpasero You're right, I've updated the title to make this more clear. However, I'm not sure it's "unrelated to general JSON files" in VS Code: when I edit any JSON file, comments are allowed.

**Comment by bpasero** [comment]
I would argue this is a bug. We should only allow our custom JSON (with comments etc) in our settings.

**Comment by borekb** [comment]
I also think it would be best if there would be a clear distinction between standard JSON and "VS Code Config JSON".

There are actually projects like http://json5.org/ but I guess the original VSCode implementation didn't want to deal with all the possible complexities of JSON5 and "just" added comments to JSON (which, in turn, created issues of its own).

**Comment by aeschli** [comment]
When reading the setting files, we already ignore the trailing commas. But there is still a warning, and I think that should stay.

**Comment by borekb** [comment]
It's an error, not a warning. Would be nice if this was consistent between comments and commas.

**Comment by bbaia** [comment]
+1 for adding an option.
Formattting removes the trailing comma.

When I add a new parameter in a config JSON file, there is 2 modifications:
```javascript
{
  "Property1": true
}
```
```javascript
{
  "Property1": true, // 1) trailing comma on the previous parameter
  "Property2": false // 2) new line
}
```

This option will avoid some git conflicts

**Comment by Venryx** [comment]
+ for adding an option. I fairly often move settings around for better organization, and it's a pain to always add a comma for items moved from the end, and remove a comma for items moved to the end.

**Comment by Quasilyte** [comment]
+1. I do not know any person who likes to manually keep JSON in pedantically "valid" state.

Maybe VS Code could automatically fix those errors on save?

**Comment by AbsoluteZero273** [comment]
How do I disable a linter for json?

**Comment by aeschli** [comment]
Turning off the JSON validator is currently not possible.

**Comment by aeschli** [comment]
duplicate of #26830, fixed in 1.19. We will no longer report trailig commas in our settings files.
Regular JSON files (package.json...) will still get the error.
